fontique/src/backend/coretext.rs

```diff
@@ -77,97 +71,39 @@ impl SystemFonts {
     pub(crate) fn fallback(&mut self, key: impl Into<FallbackKey>) -> Option<FamilyId> {
         let key = key.into();
         let sample = key.script().sample()?;
-        self.fallback_for_text(sample, key.locale(), false)
+        let font = self.fallback_font_for_text(sample, key.locale(), false)?;
+        let family_name = unsafe { CTFontCopyFamilyName(&font) };
+        self.name_map.get(&family_name.to_string()).map(|n| n.id())
     }
-}
 
-impl SystemFonts {
-    fn fallback_for_text(
+    fn fallback_font_for_text(
         &mut self,
         text: &str,
         locale: Option<&str>,
         prefer_ui: bool,
-    ) -> Option<FamilyId> {
-        let font = fallback_for_text(text, locale, prefer_ui)?;
-        self.name_map.get(&font.family_name()).map(|n| n.id())
-    }
-}
-
-fn fallback_for_text(text: &str, locale: Option<&str>, prefer_ui: bool) -> Option<CTFont> {
-    let cf_text = CFString::new(text);
-    let cf_text_range = CFRange::init(0, cf_text.char_len());
-    let cf_locale = locale.map(CFString::new);
-    let base_font = {
-        let desc_attrs = CFDictionary::from_CFType_pairs(&[]);
-        let mut desc = font_descriptor::new_from_attributes(&desc_attrs);
-        if prefer_ui {
-            if let Some(ui_desc) = ui_font_for_language(kCTFontSystemFontType, 0.0, None)
-                .map(|font| font.copy_descriptor())
-            {
-                desc = ui_desc;
-            }
-        }
```

Comment:
I didn't think we needed to get the UI front, then get the descriptor from that, then get a font from that descriptor. If we want a UI font, can't we just use the UI font that get from `CTFontCreateUIFontForLanguage`?

Reply:
That seems plausible, not sure why it's done that way right now.

Reply:
At some point, this was more involved and tried to select a font with specific properties thus a descriptor was required. The simplification looks good to me.